rest/recorder.go

```diff
@@ -43,6 +43,9 @@ type recorderResponseWriter struct {
 
 // Record the status code.
 func (w *recorderResponseWriter) WriteHeader(code int) {
+	if w.wroteHeader {
+		return
+	}
 	w.ResponseWriter.WriteHeader(code)
```

Comment:
Shouldn't this line be before the test on wroteHeader ? Just in case the underlying net/http implementation (or future implementation) does something even if the header is already written.